src/main.rs

```diff
@@ -76,6 +68,49 @@ use smp::PerCpu;
 use sync::Once;
 use umode::UmodeTask;
 
+#[cfg(test)]
+mod tests {
```

Comment:
nit: can we put this `mod` at the bottom of the file so it's out of the way?

Reply:
fixed

Reply:
fixed